### PR TITLE
R2s frontend

### PR DESCRIPTION
**Pull request by elliottbiondo**
This PR creates a CLI for the in capabilities in the `r2s` module. I will make a user manual entry for this too, which I will cite from this script.

### CODE REVIEW

**Comment by scopatz** [comment]
Should we wait to merge this until the user manual entry, then?

File: scripts/r2s.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#!/bin/env python
+import argparse
+import ConfigParser
```
Comment (scopatz) [inline scripts/r2s.py:3]: I didn't know anyone used configparse anymore...

File: scripts/r2s.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    subparsers = parser.add_subparsers(help=r2s_help, dest='command')
+
+    setup_parser = subparsers.add_parser('setup', help=setup_help)
+    step1_parser = subparsers.add_parser('step1', help=step1_help)
+    step2_parser = subparsers.add_parser('step2', help=step2_help)
+
+    args, other = parser.parse_known_args()
+    if args.command == 'setup':
+        setup()
+    if args.command == 'step1':
```
Comment (scopatz) [inline scripts/r2s.py]: Shouldn't this be `elif`?

**Comment by scopatz** [comment]
I think it would be nice to have a command line option to specify the output file. like a `-o` or a `-f`.  Also, I think the the functions here should have some minimal docstrings.  Other than that it looks great.

File: scripts/r2s.py (hunk 2 above)
Comment (elliottbiondo) [inline scripts/r2s.py]: Yep! Fixed.

**Comment by elliottbiondo** [comment]
Sure, I can just do the user manual now.

In regards to the output files, I am not sure changing the names of files would be too useful. `step1` produces: `alara_geom`, `alara_matlib`, `alara_fluxin`, and the latter two all have to match what is specified in `alara_params`. `step_2` produces `source_1.h5m`, `source_2.h5m` ... `source_N.h5m` and these files all must be eventually changed to `source.h5m` for photon transport due to an MCNP limitation. Each usage of r2s.py is meant to be done in a different folder which I will explain in the user manual.

**Comment by scopatz** [comment]
My concern is that hard coding paths like this prevents best practices such as naming the file after uuids or some other experiment run number. If all files have the same name but are supposed to be in uuid'd directories, this doesn't scale super well in my experience because invariably files get moved around and the provenance of the original file is lost. 

If something needs to be named source.h5m eventually, that should be done as a copy or softlink of the original uuid'd file.

I'm happy for the file names that you chose to be the defaults. They make sense. However, forcing people to use these names is a little problematic, very easy to fix as a developer, and very difficult to fix as a user. I hope this makes sense...

**Comment by elliottbiondo** [comment]
Got it. I just added an option to rename the output.

**Comment by scopatz** [comment]
Thanks @elliottbiondo

**Pull request merged**